shared/src/main/scala/amf/core/AMFGraphPartialCompiler.scala

```diff
@@ -0,0 +1,98 @@
+package amf.core
+
+import amf.client.remod.amfcore.plugins.parse.AMFParsePlugin
+import amf.client.remod.amfcore.plugins.{HighPriority, PluginPriority}
+import amf.core.errorhandling.AMFErrorHandler
+import amf.core.exception.UnsupportedParsedDocumentException
+import amf.core.metamodel.Obj
+import amf.core.metamodel.Type.ObjType
+import amf.core.metamodel.document.{BaseUnitModel, DocumentModel}
+import amf.core.metamodel.domain.DomainElementModel
+import amf.core.model.document.{BaseUnit, Document}
+import amf.core.model.domain.{AmfObject, DomainElement}
+import amf.core.parser.{Annotations, Fields, ParserContext, ReferenceHandler, SyamlParsedDocument}
+import amf.plugins.document.graph.parser.{
+  FlattenedGraphParser,
+  FlattenedUnitGraphParser,
+  GraphDependenciesReferenceHandler,
+  GraphParserContext
+}
+
+class AMFGraphPartialCompiler(compilerContext: CompilerContext, startingPoint: String)
+    extends AMFCompiler(compilerContext, Some("application/graph+json")) {
```

Comment:
Why inheritances instead of composition? Can't we instantiate an AMFCompiler with only one plugin?

Reply:
To use the same flow of seaching syntax plugins and just modify the part that I need to hack

Reply:
but we will always use the same syntax plugin, syaml

Reply:
For know, in a future maybe now. I like the idea to be prepared.
Even if is always syaml, all the invocation context of the compiler can be reused